### PR TITLE
EVG-19619 Add nil check for projectRef

### PR DESCRIPTION
**Pull request by bynn**
EVG-19619

### Description
add nil check for projectRef

### Testing
staging 
called api without panic

### CODE REVIEW

File: service/api.go [hunk 1]
```diff
@@ -263,6 +264,12 @@ func (as *APIServer) validateProjectConfig(w http.ResponseWriter, r *http.Reques
 
 	if input.Quiet {
 		errs = errs.AtLevel(validator.Error)
+	} else if projectRef != nil || !isConfigDefined {
```
Comment (Kimchelly) [inline service/api.go:267]: Is this supposed to be `projectRef == nil`?

Comment (Kimchelly) [reply service/api.go:267]: Also, should this skip alias validation when there's no project config? I thought that the project config was optional.

**Review by Kimchelly — CHANGES_REQUESTED** [review]

File: service/api.go (hunk 1 above)
Comment (bynn) [reply service/api.go:267]: you are so right

Comment (hadjri) [reply service/api.go]: `ConstructMergedAliasesByPrecedence` is meant to return a list of aliases regardless of whether the project config is defined so yes, we'd want to do alias validation regardless

**Review by Kimchelly — APPROVED** [review]

**Pull request merged**